src/engraving/dom/tie.cpp

```diff
@@ -213,6 +212,148 @@ Tie::Tie(const ElementType& type, EngravingItem* parent)
     setAnchor(Anchor::NOTE);
 }
 
+void Tie::collectPossibleEndPoints()
+{
+    const MasterScore* master = masterScore();
+    const Note* note = toNote(parentItem());
+    const Chord* chord = note->chord();
+    const Measure* measure = chord->measure();
+    const MeasureBase* masterMeasureBase = master->measure(measure->index());
+    const Measure* masterMeasure = masterMeasureBase && masterMeasureBase->isMeasure() ? toMeasure(masterMeasureBase) : nullptr;
+    if (!tieEndPoints()) {
+        return;
+    }
+
+    std::vector<Tie*> oldTies;
+    for (TieEndPoint* endPoint : *tieEndPoints()) {
+        oldTies.push_back(endPoint->endTie());
+    }
+
+    tieEndPoints()->clear();
+
+    if (!startNote()->followingJumpItem()) {
+        return;
+    }
+
+    int endPointIdx = 0;
+
+    Note* nextNote = searchTieNote(note);
+    nextNote = nextNote ? nextNote : endNote();
+
+    if (nextNote) {
+        const bool hasTie = nextNote->tieBack();
+        TieEndPoint* endPoint = new TieEndPoint(nextNote, u"next note", hasTie, endPointIdx, true);
+        tieEndPoints()->add(endPoint);
+        endPointIdx++;
+    }
+
+    // Get following notes by taking repeats
+    const RepeatList& repeatList = master->repeatList();
+
+    for (auto it = repeatList.begin(); it != repeatList.end(); it++) {
+        const RepeatSegment* rs = *it;
+        const auto nextSegIt = std::next(it);
+        if (!rs->endsWithMeasure(masterMeasure) || nextSegIt == repeatList.end()) {
+            continue;
+        }
+
+        // Get next segment
+        const RepeatSegment* nextSeg = *nextSegIt;
+        const Measure* firstMasterMeasure = nextSeg->firstMeasure();
+        const MeasureBase* firstMeasureBase = firstMasterMeasure ? score()->measure(firstMasterMeasure->index()) : nullptr;
+        const Measure* firstMeasure = firstMeasureBase && firstMeasureBase->isMeasure() ? toMeasure(firstMeasureBase) : nullptr;
+        const Segment* firstCrSeg = firstMeasure ? firstMeasure->first(SegmentType::ChordRest) : nullptr;
+        if (!firstCrSeg) {
+            continue;
+        }
+
+        Note* nextNote = searchTieNote(note, firstCrSeg);
+
+        if (nextNote) {
+            bool hasIncomingTie = nextNote->tieBack();
+            String jumpName = nextNote->precedingJumpItemName();
+            TieEndPoint* endPoint = new TieEndPoint(nextNote, jumpName, hasIncomingTie, endPointIdx, false);
+            tieEndPoints()->add(endPoint);
+            endPointIdx++;
+        }
+    }
+    for (Tie* tie : oldTies) {
+        auto findEndTie = [&tie](const TieEndPoint* endPoint){
+            return endPoint->endTie() == tie;
+        };
+
+        if (std::find_if((*tieEndPoints()).begin(), (*tieEndPoints()).end(), findEndTie) != (*tieEndPoints()).end()) {
+            continue;
+        }
+        score()->undoRemoveElement(tie);
```

Comment:
A method of the Tie class should not be removing other ties from the Score, it could be a bit dangerous. Also, from a function named collectPossibleEndPoints I'd probably expect something which returns me a list of TieEndPoint*, not something which actively modifies the current tie and even the score. Here I would try to do a combination of (not necessarily all of them)

- Giving this method a more descriptive name
- Moving out the code which removes other ties into a separate function
- Given that this method performs an undoable action on the score, it looks more like the kind of methods we usually put in edit.cpp, that's also an option

Reply:
Yes, this code to remove ties is absolutely in the wrong place - it doesn't even cover all the situations I was hoping.  
I think we need to remove stale endpoints when the repeat structure is recalculated - either inside or just after `RepeatList::unwind`.